tools/utils.py

```diff
@@ -263,3 +266,95 @@ def normalize_image(
     b = target_max - a * image.max()
     image_norm = (a * image + b).astype(target_type)
     return image_norm
+
+
+class BoundaryDataset(Dataset):
+    def __init__(self,
+                 img_paths: List[str],
+                 input_size: Union[int, List[int]] = (1024, 1024)):
+        self.img_paths = img_paths
+        self.input_size = (input_size, input_size) if isinstance(input_size, int) else input_size
+
+    def __len__(self):
+        return len(self.img_paths)
+
+    def __getitem__(self, idx):
+        img_path = self.img_paths[idx]
+        img = cv2.imread(img_path)
+        img = cv2.resize(img, self.input_size)
+        return img, img_path
+
+
+class MorphologicalTransformations:
+    def __init__(
+            self,
+            image_file: str,
+    ):
+
+        self.image_file = image_file
+        self.MAX_PIXEL = 255
+        self.MIN_PIXEL = 0
+
+    def read_image(self):
```

Comment:
@sriva-e 

Why do we have to read images, if we assume that the input of the binary method is ndarray image?